mmtrack/datasets/base_sot_dataset.py

```diff
@@ -0,0 +1,285 @@
+import os.path as osp
+import random
+from abc import ABCMeta, abstractmethod
+
+import numpy as np
+from mmcv.utils import print_log
+from mmdet.datasets.pipelines import Compose
+from torch.utils.data import Dataset
+
+from mmtrack.core.evaluation import eval_sot_ope
+from mmtrack.datasets import DATASETS
+
+
+@DATASETS.register_module()
+class SOTDataset(Dataset, metaclass=ABCMeta):
+    """Dataset of single object tracking.
+
+    The dataset can both support training and testing mode.
+    """
```

Comment:
complete the docstring for the `class`. you can refer to this [example](https://github.com/open-mmlab/mmtracking/blob/master/mmtrack/models/trackers/masktrack_rcnn_tracker.py#L14)